language-support/ts/daml-ledger/index.ts

```diff
@@ -127,15 +131,15 @@ const decodeArchiveEventUnknown: jtv.Decoder<ArchiveEvent<object>> =
  * Decoder for an [[Event]].
  */
 const decodeEvent = <T extends object, K, I extends string>(template: Template<T, K, I>): jtv.Decoder<Event<T, K, I>> => jtv.oneOf<Event<T, K, I>>(
-  jtv.object({created: decodeCreateEvent(template)}),
+  jtv.object({created: decodeCreateEvent(template), matchedQueries: jtv.array(jtv.number())}),
   jtv.object({archived: decodeArchiveEvent(template)}),
```

Comment:
Is it a JSON API limitation that you can't match archive events to specific streams?

Reply:
Unfortunately yes. When reading the event from the Ledger API, creates are matched with the predicate based on the contract arguments. The archive event doesn't have the arguments, so this cannot be done. I guess it should be possible to achieve this by matching up the active contract set with the archive event and retrieving the argument from there, but this is not currently possible with the JSON API.